css/src/components/checkbox.scss

```diff
@@ -0,0 +1,121 @@
+/* stylelint-disable selector-max-specificity, max-nesting-depth, selector-no-qualifying-type */
+
+$checkbox-border-radius: $radius-sm !default;
+$checkbox-border-color: $text !default;
+$checkbox-check-hover-color: $secondary !default;
+$checkbox-icon-size: 0.875em !default;
+$checkbox-icon-size-small: 0.625em !default;
+$checkbox-size: 1.25em !default;
+$checkbox-size-small: 0.875em !default;
+$checkbox-timing-function: $input-timing-function !default;
+$checkbox-duration: $input-transition-duration !default;
+$checkbox-spacing: $spacer-3;
+
+.checkbox {
+	display: flex;
+	position: relative;
+	align-items: center;
+	line-height: 1.25;
+	cursor: pointer;
+
+	+ .checkbox {
+		margin-top: $checkbox-spacing;
+		margin-left: 0;
+	}
+
+	.checkbox-check {
+		display: flex;
+		flex-shrink: 0;
+		align-items: center;
+		justify-content: center;
+		width: $checkbox-size;
+		height: $checkbox-size;
+		transition-duration: $checkbox-duration;
+		transition-property: background, border, border-color;
+		transition-timing-function: $checkbox-timing-function;
+		border: 1px solid $checkbox-border-color;
+		border-radius: $checkbox-border-radius;
+
+		&:not(:last-child) {
+			margin-#{$user-right}: 0.5rem;
+		}
+
+		&::before {
+			font-size: $checkbox-icon-size;
+			content: '\0020';
+		}
+	}
+
+	&:hover {
+		.checkbox-check::before {
+			@include checkmark;
+		}
+	}
+
+	&.checkbox-sm {
```

Comment:
after the `em` sizing changes, can't we just use `.checkbox-sm { font-size: $font-size-7; }`?

Reply:
(and cut the rest)

Reply:
Not quite. At `.checkbox-sm`, 1.25em size equates to 17.5px, which is too large so the unit needs to be adjusted to 1em. Updated to `$font-size-8` (14px) and removed extra `.checkbox-text` rule.

Reply:
Updated so that `1em` size works for both.

Reply:
Toyed around with this a bit locally.

```
	&.checkbox-sm {
		font-size: 14px;
	}
```

Is sufficient to resize the whole box, and keep things in proportion. Need to update borders to use `em` sizing too.

Reply:
Updated